include/boost/url/detail/impl/pct_format.ipp

```diff
@@ -0,0 +1,215 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#ifndef BOOST_URL_DETAIL_IMPL_PCT_FORMAT_IPP
+#define BOOST_URL_DETAIL_IMPL_PCT_FORMAT_IPP
+
+#include <boost/url/detail/pct_format.hpp>
+
+namespace boost {
+namespace urls {
+namespace detail {
+
+std::size_t
+pct_vmeasure(
+    grammar::lut_chars const& cs,
+    format_parse_context& pctx,
+    measure_context& mctx)
```

Comment:
are these two contexts always passed together?

Reply:
It's sometimes `format_parse_context` and `format_context` as well.